rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -592,6 +602,89 @@ private void saveCoinbaseInformations() {
             safeSaveToRepository(getStorageKeyForCoinbaseInformation(blockHash), data, BridgeSerializationUtils::serializeCoinbaseInformation));
     }
 
+    public Long getActiveFederationCreationBlockHeight() {
+        if (!activations.isActive(RSKIP186)) {
+            return null;
+        }
+
+        if (activeFederationCreationBlockHeight != null) {
+            return activeFederationCreationBlockHeight;
+        }
+
+        activeFederationCreationBlockHeight = safeGetFromRepository(ACTIVE_FEDERATION_CREATION_BLOCK_HEIGHT_KEY, BridgeSerializationUtils::deserializeOptionalLong).orElse(null);
+        return activeFederationCreationBlockHeight;
+    }
+
+    public void setActiveFederationCreationBlockHeight(long activeFederationCreationBlockHeight) {
+        this.activeFederationCreationBlockHeight = activeFederationCreationBlockHeight;
+    }
+
+    protected void saveActiveFederationCreationBlockHeight() {
+        if (activeFederationCreationBlockHeight == null || !activations.isActive(RSKIP186)) {
+            return;
+        }
+
+        safeSaveToRepository(ACTIVE_FEDERATION_CREATION_BLOCK_HEIGHT_KEY, activeFederationCreationBlockHeight, BridgeSerializationUtils::serializeLong);
+    }
+
+    public Long getNextFederationCreationBlockHeight() {
+        if (!activations.isActive(RSKIP186)) {
+            return null;
+        }
+
+        if (nextFederationCreationBlockHeight != null) {
+            return nextFederationCreationBlockHeight;
+        }
+
+        nextFederationCreationBlockHeight = safeGetFromRepository(NEXT_FEDERATION_CREATION_BLOCK_HEIGHT_KEY, BridgeSerializationUtils::deserializeOptionalLong).orElse(null);
+        return nextFederationCreationBlockHeight;
+    }
+
+    public void setNextFederationCreationBlockHeight(long nextFederationCreationBlockHeight) {
+        this.nextFederationCreationBlockHeight = nextFederationCreationBlockHeight;
+    }
+
+    public void clearNextFederationCreationBlockHeight() {
+        this.nextFederationCreationBlockHeight = -1L;
```

Comment:
It's not clear that the `BridgeSerializationUtils` used for this variable could manage a negative value (my first look: it uses `BigInteger.fromUnsignedByteArray`). But then it uses `BigInteger.longValue` ;  do we have this test?

Reply:
`-1L` value is not being serialized. `null` is being stored instead. It's just some sort of flag that indicates that the value should be cleared during the `save` stage.